gateway-messages/src/sp_to_mgs/ignition.rs

```diff
@@ -0,0 +1,168 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Types describing ignition state; see RFD 141.
+
+use core::fmt;
+use hubpack::SerializedSize;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::tlv;
+
+// Confirm our expectation that for our current product (35 ports), we can fit
+// the full bulk ignition state into a single UDP packet.
+static_assertions::const_assert!(
+    IgnitionState::MAX_SIZE * 35 <= crate::MIN_TRAILING_DATA_LEN
+);
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, SerializedSize, Serialize, Deserialize,
+)]
+pub enum IgnitionError {
+    FpgaError,
+    InvalidPort,
+    InvalidValue,
+    NoTargetPresent,
+    RequestInProgress,
+    RequestDiscarded,
+    Other(u32),
+}
+
+impl fmt::Display for IgnitionError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = match self {
+            IgnitionError::FpgaError => "fpga communication error",
+            IgnitionError::InvalidPort => "invalid target",
+            IgnitionError::InvalidValue => "invalid value",
+            IgnitionError::NoTargetPresent => "no target present",
+            IgnitionError::RequestInProgress => "request in progress",
+            IgnitionError::RequestDiscarded => "request discarded",
+            IgnitionError::Other(code) => {
+                return write!(f, "other (code = {code})");
+            }
+        };
+        write!(f, "{s}")
+    }
+}
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, SerializedSize, Serialize, Deserialize,
+)]
+pub struct IgnitionState {
+    pub receiver_status: ReceiverStatus,
+    pub target: Option<Target>,
+}
+
+impl IgnitionState {
+    pub const TAG: tlv::Tag = tlv::Tag(*b"IGN0");
+}
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, SerializedSize, Serialize, Deserialize,
+)]
+pub struct ReceiverStatus {
+    pub aligned: bool,
+    pub locked: bool,
+    pub polarity_inverted: bool,
+}
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, SerializedSize, Serialize, Deserialize,
+)]
+pub struct Target {
```

Comment:
Nit: Everywhere else in this PR, `target` is a destination encoded as a `u8`. This is slightly confusing, although in context it would be clear. Maybe rename this type to `TargetState` to better reflect what it is.